From c09bf19de0fdb2fb0ad64317953eeacc48ede15e Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Wed, 23 Jul 2025 10:38:39 -0300
Subject: [PATCH] Fix non-POSIX paths in Libdir under Windows

This fixes using the GStreamer plugins .pc files with e.g. CMake, which
interprets backslashes as an escape code and strips them automatically.
---
 src/pkg_config_gen.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pkg_config_gen.rs b/src/pkg_config_gen.rs
index 98e3017..8e0c030 100644
--- a/src/pkg_config_gen.rs
+++ b/src/pkg_config_gen.rs
@@ -111,7 +111,7 @@ impl PkgConfig {
         }
 
         let libs = vec![
-            format!("-L{}", libdir.display()),
+            format!("-L{}", canonicalize(libdir.display().to_string())),
             format!("-l{}", capi_config.library.name),
         ];
 
